### PR TITLE
test: handle async action scenarios

### PR DESCRIPTION
**Pull request by joaqcid**
I - Promise that returns an observable
II - Observable that returns a promise

## PR Type

```
[ ] Other... Please describe: Test covering existing scenario
```
Issue Number: [#1660](https://github.com/ngxs/store/issues/1660)

## Does this PR introduce a breaking change?

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: integrations/hello-world-ng9-ivy/package.json [hunk 1]
```diff
@@ -43,8 +43,8 @@
     "angular-http-server": "file:../../node_modules/angular-http-server",
     "core-js": "file:../../node_modules/core-js",
     "cypress": "file:../../node_modules/cypress",
-    "jest": "file:../../node_modules/jest",
-    "jest-preset-angular": "file:../../node_modules/jest-preset-angular",
+    "jest": "25.5.4",
```
Comment (splincode) [inline integrations/hello-world-ng9-ivy/package.json:46]: what do you think about install correct jest in root package.json?

**Review by splincode — APPROVED** [review]

**Comment by splincode** [comment]
@markwhitfeld any news?)

**Comment by splincode** [comment]
@arturovt what do you mean about this?

**Comment by splincode** [comment]
@arturovt do can we merge it?

**Review by markwhitfeld — APPROVED** [review]
This documents the existing behaviour, which has a bug.
I wanted to change some test names, but this branch is in your repo.
I'll adjust it on master

**Comment by markwhitfeld** [comment]
This has been fixed as part of the [v3.7.4](https://github.com/ngxs/store/releases/tag/v3.7.4) release.

**Pull request merged**